### ISSUE TITLE
watching useFieldArray fields don't always re-render when changes happen

### ISSUE THREAD

**Issue by matheusffatg** [react-hook-form/react-hook-form#981]
**Describe the bug**
when watching useFieldArray, it doesn't always re-render.
It happens after appending a new item to the field array.
![image](https://user-images.githubusercontent.com/59067561/73875956-f5447a80-480a-11ea-9e2c-4a3ee305adcc.png)

**To Reproduce**
1. Go to sandbox https://codesandbox.io/s/react-hook-form-usefieldarray-0q2eu
2. Append 1 item (item1)
3. type anything
      * note span gets updated
4. Append another item (item2)
5. type anything on the new item
      * note span does not update
6 . type on item1. 
      * note span all spans updating
7 . type on last item2
      * note span gets updated

**Codesandbox link**
https://codesandbox.io/s/react-hook-form-usefieldarray-0q2eu

**Expected behavior**
it should re-render when watching for useFieldArray and the field changes

**Comment by bluebill1049** [comment]
That's actually the intended behavior. here is how `watch` works.

`watch` is already 1 tick behind the render and also starts reading value once user starts typing. That's the difference between local state.  unless you can provide a default value for the `watch`

**Comment by matheusffatg** [comment]
yeah I did read somewhere that `watch` is 1 tick behind and sounds like people are confused about it :D

> unless you can provide a default value for the watch

on the example I create, the default value is null.
```
<Controller
       // ...
        defaultValue={item.custom ? item.custom.name : null}
```
the watch also has a default value in case any change happens:
```
const watchName = watch(`test[${index}].custom.name`, item.custom ? item.custom.name : null);
```
or you mean the `defaultValues` when from `useForm`?

I guess the real scenario here is that this array field is an object (e.g. user, so field name would be either `user.id` or `user_id`)
how would you go about  watching this field to make sure once the selection happens, the table updates?
for now, here is what I am doing.
```
<Controller
      // ...
      onChange={([selection]: [IUser]): any => {
            setValue(`${field}.user.id`, selection.id);
            return { value: selection.id };
      }
```

**Comment by bluebill1049** [comment]
try to watch the arrayField name `watch('arrayField')`

**Comment by seonghyeonkimm** [comment]
When I append new fields using `useFieldArray ` hook, it doesn't trigger rerender the component having watch function. Is this also expected behavior? I need to trigger rerender when array's length is changed. Is there another way to do it? @bluebill1049

**Comment by bluebill1049** [comment]
https://react-hook-form.com/api#useFieldArray plz read the important part about `watch`
